src/query/plan/operator.cpp

```diff
@@ -1578,17 +1586,26 @@ class SingleSourceShortestPathCursor : public query::plan::Cursor {
             throw QueryRuntimeException("Expansion condition must evaluate to boolean or null.");
         }
       }
-      to_visit_next_.emplace_back(edge, vertex);
+      to_visit_next_.emplace_back(edge, vertex, curr_acc_path);
       processed_.emplace(vertex, edge);
     };
 
     // populates the to_visit_next_ structure with expansions
     // from the given vertex. skips expansions that don't satisfy
     // the "where" condition.
-    auto expand_from_vertex = [this, &expand_pair](const auto &vertex) {
+    auto expand_from_vertex = [this, &pull_memory, &frame, &expand_pair](const auto &vertex) {
       if (self_.common_.direction != EdgeAtom::Direction::IN) {
+        Path curr_acc_path(pull_memory);
```

Comment:
I think instead of using pull memory, you could use frame memory since each object on frame is instantiated with "frame" memory. Because when we are doing move with objects using different memory we will actually have copy